packages/jest-haste-map/src/crawlers/node.js

```diff
@@ -0,0 +1,57 @@
+ /**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+'use strict';
+
+const denodeify = require('denodeify');
+const fs = require('graceful-fs');
+const glob = require('glob');
+const path = require('../fastpath');
+
+const statFile = denodeify(fs.stat);
+const globPromise = denodeify(glob.bind(glob));
+
+module.exports = function nodeCrawl(roots, extensions, ignorePattern, data) {
+  const files = Object.create(null);
+  const pattern =
+    path.sep + '**' + path.sep + '*.{' + extensions.join(',') + '}';
+
+  return Promise.all(roots.map(
+    root => globPromise(root + pattern)
```

Comment:
this nice and simple implementation of the node crawler actually turns out to have 2x worse performance than the one in node-haste. I might revert to that one instead. Didn't think glob would be sooo slow.